File: app/src/main/java/com/example/ciclomenstrual/database/NoteDao.java
package com.example.ciclomenstrual.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.ciclomenstrual.database.Note;

import java.util.List;

@Dao
public interface NoteDao {
    @Query("SELECT * FROM notes WHERE date = :date")
    List<Note> getNotesForDate(long date);

    @Query("SELECT * FROM notes")
    List<Note> getAllNotes();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Note note);

    @Delete
    void delete(Note note);

    @Query("DELETE FROM notes WHERE date = :date AND content = :content")
    void deleteNoteByDateAndContent(long date, String content);
}

File: app/src/main/java/com/example/ciclomenstrual/data/local/NoteDao.kt
package com.example.ciclomenstrual.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface NoteDao {
    @Query("SELECT * FROM notes")
    suspend fun getAll(): List<RoomNote>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(note: RoomNote): Long

    @Query("DELETE FROM notes WHERE id = :id")
    suspend fun deleteById(id: Long)
}
